Don't run CI pipeline on pull requests that change documentation only

just95 opened the pull request:
Since the master branch is now protected, updates of the README and documentation have to be performed via pull requests, too. In order to avoid wasting resources, I suggest that the CI pipeline no longer runs on pull requests that modify only the README or other Markdown files only. I've modified the GitHub actions workflow configuration file such that the same rules that applied to pushes before now apply to pull requests as well.

just95 committed:
Skip CI pipeline on pull requests that modify documentation only

Since the master branch is now protected, updates of the README
and documentation have to be performed via pull requests. In order
to avoid wasting resources, the CI pipeline will no longer run
on pull requests that modify only the README or other Markdown
files. The same rules that applied to pushes before, now apply
to pull requests as well.

just95 commented:
Another interesting observation: Even if the CI pipeline ran for a commit on a push event already, it apparently runs on a pull request again (see above :sweat_smile:). @ichistmeinname do you think that I should disable the CI pipeline on push events entirely?

ichistmeinname commented:
Since all checks (i.e. the pipeline) were running for this pull request already, I guess that we can then disable to rerun them when the request merged. @nbun Do you agree?

ichistmeinname reviewed (changes requested):
Remove GitHub action for push events.

just95 committed:
Skip CI pipeline on push events #4

Since it is not possible to push to the master branch directly
and the CI pipeline runs on pull request events anyway, it is
not neccessary to run the CI pipeline on every push. If it did
it would be executed at least twice for every pull request which
would be a waste of resources.
